### ISSUE TITLE
Metric Removal Breaks Report Builder

### ISSUE THREAD

**Issue by balajimanoharan** [yahoo/navi#86]
Problem:
If a metric is removed from the WS metadata, the report builder breaks and there is no way to recover the report. 

Solution:
We should ideally clean up the removed metrics from the report object and notify the owner that a metric from the report has been removed, but still allow the user to view the report or modify it.

**Comment by balajimanoharan** [comment]
Even if we do not have a messaging system, it would be nice to recover the report and have the invalid metric/dimension removed automatically. We had an issue with Digits recently where we had to manually remove a metric from every report in the DB, because we removed it from the WS. It would have been great if the UI could recover automatically from a removal.

**Comment by cythrawll** [comment]
I agree it would be great if we could auto do this.  But I think we need to make sure that the user knows that's what is happening.

I'm not sure a user would like it if their reports were slowly loosing metrics over time.

**Comment by balajimanoharan** [comment]
What I meant was we could just utilize the exisiting notification service to notify the users. The notification would just show up when we remove the metric, and not every time the report is visited. We don't really need the persistent messaging system. Once the metric has been removed and the updated record persisted back, there would be no need for any further updates.
